### PR TITLE
Use the Unity PAL for the IL2CPP debugger

### PR DESCRIPTION
**Pull request by joshpeterson**
Change the IL2CPP manager debugger to use the Unity PAL, so the debugger
can work on all platforms that IL2CPP supports.

### CODE REVIEW

**Comment by mderoy** [comment]
@joshpeterson if we're using the PAL for il2cpp now, can we remove the "-win32", "-mac" etc files from the copy scripts, and all that logic about what to include from the builder?

File: mono/utils/networking-unity.c [hunk 1]
```diff
@@ -1,12 +1,86 @@
...
+    if (UnityPalGetHostByName(hostname, &name, &family, &aliases, &addresses, &addressSize) == kWaitStatusSuccess)
+        add_hostent(addr_info, flags, name, family, aliases, addresses, addressSize);
+
+    g_free(name);
+    free_null_terminated_array(aliases);
+    free_null_terminated_array(addresses);
+
+    if (!addr_info->entries) {
+        *result = NULL;
+        mono_free_address_info(addr_info);
```
Comment (joncham) [inline mono/utils/networking-unity.c]: We allocate above with `g_new0` I assume this is correct de-allocator?

Comment (joshpeterson) [reply mono/utils/networking-unity.c]: Yes, this is the proper way to de-allocate. This code was taken directly from Mono.

**Comment by joshpeterson** [comment]
@mderoy:We're only using the PAL for the debugger now, so we need to keep copying those files for IL2CPP on Mono.

File: mono/metadata/w32socket-unity.c [hunk 2]
```diff
@@ -3,15 +3,26 @@
...
+#ifdef NO_HAVE_TRANSMIT_FILE_BUFFERS
+
+typedef struct {
+	gpointer Head;
+	guint32 HeadLength;
+	gpointer Tail;
+	guint32 TailLength;
+} TRANSMIT_FILE_BUFFERS;
+
+#endif
```
Comment (joshpeterson) [inline mono/metadata/w32socket-unity.c]: Note that I've just had to add this one one of our platforms. It is a bit of a hack, but I think it is acceptable for now, as it gets the build going, and it is only used in a stub method that won't be called anywhere in the debugger code.

**Comment by mderoy** [comment]
@joshpeterson whats stopping us from using the pal for il2cpp on mono? its still experimental and we'll need to do it eventually anyway.

**Comment by joshpeterson** [comment]
We're missing a number of implementations in the Unity PAL now. We could handle them, but it is just a matter of getting the time to do it.

**Pull request merged**